lib-cloudsdk/src/main/java/com/facebook/openwifi/cloudsdk/AggregatedState.java

```diff
@@ -23,75 +23,59 @@
 public class AggregatedState {
 
 	/** Rate information with aggregated fields. */
-	public static class AggregatedRate {
+	public static class Rate {
 		/**
-		 * This is the common bitRate for all the aggregated fields.
+		 * Aggregated fields bitRate
 		 */
 		public long bitRate;
 
 		/**
-		 * This is the common channel width for all the aggregated fields.
+		 * Aggregated fields chWidth
 		 */
 		public int chWidth;
 
 		/**
 		 * Aggregated fields mcs
 		 */
-		public List<Integer> mcs = new ArrayList<>();
+		public int mcs;
 
 		/** Constructor with no args */
-		private AggregatedRate() {}
+		private Rate() {}
 
-		/** Add a Rate to the AggregatedRate */
-		private void add(Rate rate) {
-			if (rate == null) {
-				return;
-			}
-			if (mcs.isEmpty()) {
-				bitRate = rate.bitrate;
-				chWidth = rate.chwidth;
-			}
-			mcs.add(rate.mcs);
-		}
-
-		/**
-		 * Add an AggregatedRate with the same channel_width to the
-		 * AggregatedRate
-		 */
-		private void add(AggregatedRate rate) {
-			if (rate == null || rate.chWidth != chWidth) {
-				return;
-			}
-			if (mcs.isEmpty()) {
-				bitRate = rate.bitRate;
-				chWidth = rate.chWidth;
-			}
-			mcs.addAll(rate.mcs);
+		/** Constructor with args */
+		private Rate(long bitRate, int chWidth, int mcs) {
+			this.bitRate = bitRate;
+			this.chWidth = chWidth;
+			this.mcs = mcs;
 		}
 	}
 
 	/**
 	 * Radio information with channel, channel_width and tx_power.
 	 */
-	public static class Radio {
+	public static class RadioConfig {
 		public int channel;
 		public int channelWidth;
 		public int txPower;
+		public String phy;
 
-		private Radio() {}
+		/** Default constructor with no args */
+		private RadioConfig() {}
 
-		public Radio(int channel, int channelWidth, int txPower) {
+		/** Constructor with args */
+		public RadioConfig(JsonObject radio) {
```

Comment:
Why are you passing in as `JsonObject`? We are trying to remove that where possible - you can define a new class for it, or pass individual fields if you have to

Reply:
I pass it in as `JsonObect` because [radio under `SSID` is of `JsonObject` type](https://github.com/Telecominfraproject/wlan-cloud-rrm/blob/88d5766bef61d4f3e190130312d074831a6f1e8f/lib-cloudsdk/src/main/java/com/facebook/openwifi/cloudsdk/models/ap/State.java#L68). Although there is a `Radio` class under `State`. it's a Radio array which contains all the Radio observed in the State

Reply:
Ah ok, we can address this later then when also addressing it in `State`.